Improve CPE Helper with new update field (#5369)
wazuh/wazuh-documentation: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst

            "reference":"http://www.securityfocus.com/bid/106928"
        }
    }

Products with version update field
----------------------------------

Another use case where vulnerabilities cannot be confirmed by the NVD due to the generic version contained in the CPE is,
for example, the vulnerability *CVE-2022-23277* for the *Microsoft Exchange Server* software product,
which also has a new field with the ``update`` version, where the cumulative version of the product is specified.

If we consult this product in the `National Vulnerability Database (CVE-2022-23277) <https://nvd.nist.gov/vuln/detail/CVE-2022-23277>`_,
we will find the following CPEs:

.. thumbnail:: ../../../images/manual/vuln-detector/nvd-exchange-server.png
    :title: Affected software for CVE-2022-23277
    :align: center
    :width: 100%

> Where we can observe, that in the CPEs the version and update fields are specified with generic versions that do not allow us

Reviewer: I'd try to reorganize this sentence to make it sound less colloquial. In addition, I don't understand the `generic versions` meaning, they seem specific versions.

to know the patch to fix the vulnerability.

To know the patches that would mitigate the vulnerability, it is necessary to consult the `Microsoft Security Update Guide <https://msrc.microsoft.com/update-guide/vulnerability/CVE-2022-23277>`_,

> where we can know the patches that fix the vulnerability for the *Microsoft Exchange Server* product.

Reviewer: ```suggestion
where we can find the patches that fix the vulnerability for the *Microsoft Exchange Server* product.
```

.. thumbnail:: ../../../images/manual/vuln-detector/msug-exchange-server.png
    :title: CVE-2022-23277 in the Microsoft Security Update Guide
    :align: center
    :width: 100%

> As in the previous use case, we will activate the ``check_hotfix`` action to automate this search,

Reviewer: ```suggestion
As explained in the previous use case, we have to include the ``check_hotfix`` action to automate this search,
```

> in addition to following the same guide used for the other use cases explained.

Reviewer: ```suggestion
as well as following the same guide used for the previous use case.
```

                    "location": "C:\\Program Files\\Microsoft\\Exchange Server\\V15\\",
                    "name": "Microsoft Exchange Server 2016 Cumulative Update 22",
                    "format": "win",
                    "version": "15.1.2375.7",
                    "vendor": "Microsoft Corporation",
                    "agent_id": "001"
                },
            ],
            "total_affected_items": 1,
            "total_failed_items": 0,
            "failed_items": [],
        },
        "message": "All specified syscollector information was returned",
        "error": 0,
    }

 2. Find the CPE program.

  To extract the CPEs from the program, we can use the information previously consulted
  on the CPEs affected by *CVE-2022-23277*. From that source, we can see exactly which CPEs
  are checked to confirm the vulnerability. These are:

  - ``cpe:2.3:a:microsoft:exchange_server:2016:cumulative_update_21:*:*:*:*:*:*``
  - ``cpe:2.3:a:microsoft:exchange_server:2016:cumulative_update_22:*:*:*:*:*:*`` (our package would match this CPE)

  We can see that the two target programs only differ in the field *update* (cumulative_update_*nº*). So we need to use

>   the ``set_update_if_product_match`` option to add the *update* field to differentiate between the CPEs, as there will be

Reviewer: three `to` in the same sentence is difficult to read, let's try something like:
```
In this case, we need to add the ``set_update_if_product_match`` action so the *update* field is included in the CPE to match the one provided by the NVD

  | Update              |                                                      | ^Microsoft Exchange Server.*Update ([0-9]+)                      | cumulative_update_$(UPDATE) = cumulative_update_22  | set_update_if_product_matches        |
  +---------------------+------------------------------------------------------+------------------------------------------------------------------+-----------------------------------------------------+--------------------------------------+
  | MSU name            |                                                      | Microsoft Exchange Server $(VERSION) Cumulative Update $(UPDATE) | Microsoft Exchange Server 2016 Cumulative Update 22 | replace_msu_name                     |
  +---------------------+------------------------------------------------------+------------------------------------------------------------------+-----------------------------------------------------+--------------------------------------+

>   The resulting dictionary entry, including more versions of *Microsoft Exchange Server*:

Reviewer: ```suggestion
Here we have the CPE helper entry needed to detect vulnerabilities of the *Microsoft Exchange Server* program:
```

  .. code-block:: json
    :class: output

    {
        "target": "windows",
        "source": {
            "vendor": [
                "^Microsoft Corporation"
            ],
            "product": [
                "^Microsoft Exchange Server"
            ],
            "version": [
                "^Microsoft Exchange Server ([0-9]{4})"
            ],
            "update":[
                "^Microsoft Exchange Server.*Update ([0-9]+)"
            ]
        },
        "translation": {
            "vendor": [
                "microsoft"
            ],
            "product": [
                "exchange_server"
            ],
            "version": [],
            "update":[
                "cumulative_update_$(UPDATE)"
            ],
            "msu_name": [
                "Microsoft Exchange Server $(VERSION) Cumulative Update $(UPDATE)"
            ]
        },
        "action": [
            "replace_vendor",
            "replace_product",
            "set_version_if_product_matches",
            "set_update_if_product_matches",
            "replace_msu_name",
            "check_hotfix"
        ]
    }

>   Alert example using this dictionary entry:

Reviewer: ```suggestion
Once applied the entry in the CPE helper and the scanner detects it, its vulnerabilities are reported:
```
